README.md

```diff
@@ -111,9 +128,16 @@ When some options are not specified and the command is run with `--no-interactio
 ![Some options specified](docs/assets/images/no-interaction-error.png)
 
 
-## Limitations
+## Option configuration
+
+The following configuration options are available per option:
 
-This has only been tested with string options set to `InputOption::VALUE_REQUIRED` - any other options are (currently) unsupported.
+| Configuration value | Description | Allowed values |
+|---------------------|-------------|----------------|
+| type | The type of value to be expected (input will be transformed) | `int`, `integer`, `float`, `bool`, `boolean`, `string` (written as strings)|
+| description | The description for the option/the text to be displayed when asking for the option | _anything_ |
+| required | Whether or not the option is required. If it's set to true, then a value MUST be entered and they will be prompted until one is provided. If set to false, a default value will be used if set, or `null` will be returned as their input. | `true` or `false` (as boolean)
```

Comment:
Just wondering why required is needed as an option, the point of the promptable stuff was to make options required anyway, trying to think of the use case for making something promptable but not required.

Would it be like a friendly "hey, you can supply this if you want?"